src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1096,6 +1096,55 @@ bool is_object_required_superset(
     return true;
 }
 
+bool is_object_dependencies_superset(
+  json::Value const& older, json::Value const& newer) {
+    // "dependencies", if present, is a dict of <property, string_array |
+    // schema>. To be compatible, each key in older has to be in newer and the
+    // values have to be of the same type and compatible.
+    // older string array needs to be a subset of newer string array.
+    // older schema needs to be compatible with newer schema
+
+    auto [maybe_skip, older_p, newer_p] = extract_property_and_gate_check(
+      older, newer, "dependencies");
+    if (maybe_skip.has_value()) {
+        return maybe_skip.value();
+    }
+    // older and newer have "dependencies"
+
+    // all dependencies in older need to carry over in newer, and need to be
+    // compatible
+    // TODO: n^2 search
```

Comment:
Is there a ticket to link/create ƒor this?

Reply:
i'll add it, but it's more a comment help if we encounter a problematic schema that stresses the runtime

Reply:
https://redpandadata.atlassian.net/browse/CORE-6982